sonar-go-checks/src/main/java/org/sonar/go/checks/ElseIfWithoutElseCheck.java

```diff
@@ -44,11 +49,11 @@ public void initialize(InitContext init) {
       }
 
       IfTree prevTree = ifTree;
-      boolean endsWithReturn = endsWithReturnBreakOrThrow(ifTree);
+      boolean endsWithReturn = endsWithReturnBreakOrPanic(ifTree);
```

Comment:
I'm not sure if we should change the behaviour of this rule 🤔  Before it raised when `panic()` was called in last `if else` statement, what we can see in `its/ruling/src/integrationTest/resources/expected/go-S126.json` change.
IMO this rule should raise issues even if one of the branch contains `throw/return/break/panic()`.

The rule is not in Sonar_Way so I didn't found any issue on "Next" and the impact is low so it can be as it is now.

Reply:
This change is just about replacing `throw` by `panic`, which could be considered the equivalent in Go (at least in this context of execution flow). Before, it simply ignored any `panic` call.
To me, the new issue raised in `its/ruling/src/integrationTest/resources/expected/go-S126.json` looks like a TP according to the rule behavior.
I agree that I don't see the why of the logic around `throw`/`return`/`break`/`panic()`, but it would remove most logic implemented in the check, which would be a significant impact, probably outside of the scope of simply migrating this rule to go :sweat_smile: